### PR TITLE
Add files from peer manager from nRF51 SDK 10.0.0

### PR DESCRIPTION
**Pull request by andresag01**
Add a subset of the files from the peer manager from nRF51 SDK 10.0.0. This
code is required to implement whitelising in the ble-nrf51822 module.

@pan- @LiyouZhou

### CODE REVIEW

**Comment by andresag01** [comment]
**NOTE:** please try to build something before merging.

**NOTE:** Some of the files I have included are just headers and I left out the actual implementation since we actually do not require this. Would you like me to include the implementation as well for completeness?

File: script/required_files.txt [hunk 1]
```diff
@@ -16,6 +16,13 @@
 	source/nordic-sdk/components/ble/common/ble_advdata.h
 	source/nordic-sdk/components/ble/common/ble_advdata_parser.c
 	source/nordic-sdk/components/ble/common/ble_advdata_parser.h
+    source/nordic-sdk/components/ble/peer_manager/id_manager.h
+    source/nordic-sdk/components/ble/peer_manager/id_manager.c
+    source/nordic-sdk/components/ble/peer_manager/peer_manager_types.h
+    source/nordic-sdk/components/ble/peer_manager/ble_gatt_db.h
+    source/nordic-sdk/components/ble/peer_manager/ble_conn_state.h
+    source/nordic-sdk/components/ble/peer_manager/sdk_mapped_flags.h
+    source/nordic-sdk/components/ble/peer_manager/peer_database.h
```
Comment (LiyouZhou) [inline script/required_files.txt]: The indentation is different, can you fix it please.

**Comment by LiyouZhou** [comment]
Apart from the indentation, I'm happy with this. Tested ble heartrate example, it compiles fine. +1

**Comment by andresag01** [comment]
@LiyouZhou I will fix the indentation and also include some extra files such as the actual implementation of the headers I already added. Even though it is not required, I think it would be good to include the implementation for completeness.

**Comment by pan-** [comment]
I'm fine with this PR, I will merge it when the indentation is fixed.

**Comment by andresag01** [comment]
@pan- @LiyouZhou Fixed indentation. Just a note, that file is completely written using tabs rather than spaces, we should change that...

**Pull request merged**